docs/installation.rst

```diff
@@ -9,16 +13,252 @@ Fantastic! Thank you!
+* ``yarn run debug-watch-server`` if you also want to modify server files (in ``src/server``)
+
+2. Mount the ``src`` folder to the bookbrainz-site service
+
+For example:
+
+::
+
+    services:
+      bookbrainz-site:
+      # 1. Change the command to run
+        command: yarn run debug
+        volumes:
+          - "./config/config.json:/home/bookbrainz/bookbrainz-site/config/config.json:ro"
+      # 2. Mount the src directory
+          - "./src:/home/bookbrainz/bookbrainz-site/src"
+.. warning::
+  Using Webpack watch mode (``yarn run debug``) results in more resource consumption (about ~1GB increased RAM usage) compared to running the standard web server.
+
+Debugging with VSCode
+*********************
+You can use VSCode to run the server or API and take advantage of its debugger, an invaluable tool I highly recommend you learn to use.
+
+This will allow you to put breakpoints to stop and inspect the code and variables during its execution, advance code execution line by line and step into function calls, instead of putting console.log calls everywhere.
+
+`Here <https://www.youtube.com/watch?v=yFtU6_UaOtA>`_ is a good introduction to debugging javascript in VSCode.
+
+There are VSCode configuration files (in the ``.vscode`` folder) for running both the server and the tests, useful in both cases to debug into the code and 
+see what is happening as the code executes. Make sure the dependencies (postgres, redis, elasticsearch) are running, and 
+you can just open the debugger tray in VSCode, select 'Launch Program' and click the button!
+
+Testing
+=======
+The test suite is built using `Mocha <https://mochajs.org/>`_ and `Chai <https://www.chaijs.com/>`_. Before running the tests, you will need to set up a ``bookbrainz_test`` database in postgres. Here are the instructions to do so:
+
+Run the following command to create and set up the ``bookbrainz_test`` database using Docker
+::
+
+    docker-compose run --rm bookbrainz-site scripts/wait-for-postgres.sh scripts/create-test-db.sh.
+
+If you are running postgres manually outside of Docker, you can set some environment variables before running the script `scripts/create-test-db.sh`
+In particular ``POSTGRES_HOST=localhost`` but you can also set ``POSTGRES_USER``, ``POSTGRES_PASSWORD`` and ``POSTGRES_DB``.
+
+Once your testing database is set up, you can run the test suite using 
+
+* To run in Docker
+::
+
+    docker-compose run --rm bookbrainz-site yarn run test 
+
+* To run locally
+::
+  
+    yarn run test 
+
+.. note::
+  You may need to adjust your ``config/test.json`` file to match your setup.
+
+Troubleshooting
+===============
+
+General
+*******
+* ``[Error: EACCES: permission denied, scandir '/root/.npm/_logs']``
+
+  There is some incompatibility with Docker for Windows The solution
+  is to **enable WSL2** (Windows Subsystem for Linux v2) and the
+  problem goes away.
+
+.. note::
+   refresh your package catalog using ``sudo apt update``
```

Comment:
This is a separate troubleshooting item, not connected to WSL.

However it feels so generic that I'm not sure it has its place here.
Maybe as a note in the manual installation page, for linux users?

Reply:
Yeah, more like a general tip even that for linux users so might as well remove it.